pkg/controller/secret/secretsync_test.go

```diff
@@ -0,0 +1,186 @@
+package secret
+
+import (
+	"bou.ke/monkey"
+	"fmt"
+	"github.com/IBM/ibm-vpc-block-csi-driver-operator/pkg/util"
+	k8v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"reflect"
+	"testing"
+)
+
+func TestSecretSyncController_translateSecret(t *testing.T) {
+	secretNamespace := "test-ns-operator"
+	secretName := "ibm-cloud-credential"
+	cmNamespace := "test-ns-cco"
+	cmName := "cloud-conf"
+	c := &SecretSyncController{}
+	type args struct {
+		cloudSecret *k8v1.Secret
+		cloudConf   *k8v1.ConfigMap
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *k8v1.Secret
+		wantErr bool
```

Comment:
All tests in this function have `want: nil` and `wantError: true`.

Would it be possible to merge `TestSecretSyncController_translateSecret` and `TestSecretSyncController_translateSecret_success`? The only difference seem to be `monkey.Patch` call.

Alternatively, having two functions is OK, but then please rename `TestSecretSyncController_translateSecret` to `TestSecretSyncController_translateSecretError` and remove `want` and `wantErr` and clean it up a bit (you don't need to call DeepEqual to check if `got` for `nil`, right?)

Reply:
Keeping test cases as 2 functions success and error as monkey.patch is using only in success scenario.